web-ui/src/components/admin/roles/Roles.jsx

```diff
@@ -62,10 +62,30 @@ const Roles = () => {
   memberProfiles?.sort((a, b) => a.name.localeCompare(b.name));
 
   useEffect(() => {
-    if (roles) {
-      setSelectedRoles(roles.map(roleObj => roleObj.role));
-    }
-  }, [roles]);
+    const url = new URL(location.href);
+    setSearchText(url.searchParams.get('searchText') ?? '');
+    const selectedRoles = url.searchParams.get('selectedRoles');
+    const roles = selectedRoles?.split(',') ?? [];
+    if (roles.length === 0) roles.push('ADMIN', 'MEMBER', 'PDL');
+    setSelectedRoles(roles);
+  }, []);
+
+  useEffect(() => {
```

Comment:
This runs every time the user changes the searchText or selectedRoles.
It updates the query parameters in the URL.